### PR TITLE
use the session-not-on-or-after field for auth expiry time

### PR DESCRIPTION
**Pull request by nsinkov**
## Changes proposed in this PR

- change to use the correct field for SAML auth expiry

## Why are we making these changes?

Currently using the assertion's not-on-or-after, but that's only 5 minutes and intended to allow for the original redirects between the IDP and waiter. If there is a limit on the auth, then it's the optional session-not-on-or-after

### CODE REVIEW

File: waiter/src/waiter/auth/saml.clj [hunk 1]
```diff
@@ -130,17 +129,16 @@
...
                           attributes))
         conditions (.getConditions assertion)
-        audiences (mapcat #(let [audiences (.getAudiences %)]
-                             (map (fn [a] (.getAudienceURI a)) audiences))
-                          (.getAudienceRestrictions conditions))]
+        authn-statements (.getAuthnStatements assertion)
+        min-session-not-on-or-after (when (not-empty authn-statements)
+                                      (apply t/min-date '(map (.getSessionNotOnOrAfter %) authn-statements)))]
     {:attrs attrs
-     :audiences audiences
```
Comment (nsinkov) [inline waiter/src/waiter/auth/saml.clj]: dropping what I don't use already

**Comment by nsinkov** [comment]
Yeah, didn't have time to do yet

File: waiter/integration/waiter/authentication_test.clj [hunk 2]
```diff
@@ -121,6 +121,11 @@
                                                                  :headers {"content-type" "application/x-www-form-urlencoded"}
                                                                  :method :post)
             _ (assert-response-status response 303)
+            cookie-fn (fn [cookies name] (some #(when (= name (:name %)) %) cookies))
```
Comment (shamsimam) [inline waiter/integration/waiter/authentication_test.clj]: Nitpick: This is a useful one for `client_tools`, maybe name it `extract-cookie`

File: waiter/integration/waiter/authentication_test.clj [hunk 3]
```diff
@@ -121,6 +121,11 @@
...
+            auth-cookie (cookie-fn cookies "x-waiter-auth")
+            _ (is (not (nil? auth-cookie)))
+            one-hour (* 60 60)
```
Comment (shamsimam) [inline waiter/integration/waiter/authentication_test.clj]: Nitpick: `one-hour-in-secs (-> 1 t/hour t/in-seconds)`, can even inline the variable in the assertion below

**Pull request merged**